benchmark/dataset_benchmark.cpp

```diff
@@ -208,4 +221,26 @@ BENCHMARK_TEMPLATE(BM_Dataset_iterate_slice_items,
                    GenerateWithDataItems<SHORT_STRING_LENGTH>, SliceZXY)
     ->Range(1 << 2, 1 << 8);
 
+template <class Gen> static void BM_Dataset_copy(benchmark::State &state) {
+  const auto d = Gen()(state.range(0), state.range(1));
+  for (auto _ : state) {
+    Dataset copy(d);
+  }
+}
+static void Args_Dataset_copy(benchmark::internal::Benchmark *b) {
+  b->RangeMultiplier(2)
+      ->Ranges({{1, 16}, {32, 64}})
+      ->Unit(benchmark::kMicrosecond);
+}
+BENCHMARK_TEMPLATE(BM_Dataset_copy, GenerateWithDataItems<SHORT_STRING_LENGTH>)
+    ->Apply(Args_Dataset_copy);
+BENCHMARK_TEMPLATE(BM_Dataset_copy, GenerateWithDataItems<LONG_STRING_LENGTH>)
+    ->Apply(Args_Dataset_copy);
+BENCHMARK_TEMPLATE(BM_Dataset_copy,
+                   GenerateWithSparseDataItems<SHORT_STRING_LENGTH>)
+    ->Apply(Args_Dataset_copy);
+BENCHMARK_TEMPLATE(BM_Dataset_copy,
+                   GenerateWithSparseDataItems<LONG_STRING_LENGTH>)
+    ->Apply(Args_Dataset_copy);
```

Comment:
Could you add some comments explaining what the parameters are? Right now the readed has to did into the individual functions to understand what the inputs mean. Some comment for some of the other benchmarks added previously.

Is there a way in googlebenchmark to add this information to the output, maybe?

Reply:
I've improved some variable names and added comments where there are multiple arguments.

Google Benchmark doesn't seem to have a more verbose means of doing this.